README.md
#Typo3 BE User Profile Pic Extension

Adds profile picture upload option for typo3 backend users.
    
##Download

You can get the Typo3 BE User Profile Pic Extension from  http://typo3.org/extensions/repository/view/be_user_profilepic

Or you can clone or download the extension files.

##License

Copyright 2013, Harish A. Licensed under the MIT license: http://www.opensource.org/licenses/mit-license.php


ChangeLog
2012-10-18  Anchu  <ccoollh@gmail.com>

    * Initial code generated with kickstarter


ext_emconf.php
<?php

########################################################################
# Extension Manager/Repository config file for ext "be_user_profilepic".
#
# Auto generated 29-06-2013 12:27
#
# Manual updates:
# Only the data in the array - everything else is removed by next
# writing. "version" and "dependencies" must not be touched!
########################################################################

$EM_CONF[$_EXTKEY] = array(
	'title' => 'Profile pic',
	'description' => 'Adds profile picture upload option for backend users.',
	'category' => 'be',
	'author' => 'Harish a',
	'author_email' => 'harishanchu@gmail.com',
	'shy' => '',
	'dependencies' => 'taskcenter',
	'conflicts' => '',
	'priority' => '',
	'module' => '',
	'state' => 'beta',
	'internal' => '',
	'uploadfolder' => 1,
	'createDirs' => '',
	'modify_tables' => '',
	'clearCacheOnLoad' => 0,
	'lockType' => '',
	'author_company' => 'phases.dk',
	'version' => '1.0.1',
	'constraints' => array(
		'depends' => array(
			'taskcenter' => '',
		),
		'conflicts' => array(
		),
		'suggests' => array(
		),
	),
	'_md5_values_when_last_written' => 'a:10:{s:9:"ChangeLog";s:4:"9a3d";s:12:"ext_icon.gif";s:4:"90b9";s:14:"ext_tables.php";s:4:"bf79";s:14:"ext_tables.sql";s:4:"4539";s:16:"locallang_db.xml";s:4:"68f0";s:10:"README.txt";s:4:"ee2d";s:19:"doc/wizard_form.dat";s:4:"ccf0";s:20:"doc/wizard_form.html";s:4:"7057";s:47:"modfunc1/class.tx_beuserprofilepic_modfunc1.php";s:4:"4b1b";s:22:"modfunc1/locallang.xml";s:4:"e4d9";}',
	'suggests' => array(
	),
);

?>

ext_tables.php
<?php
if (!defined('TYPO3_MODE')) {
	die ('Access denied.');
}
$tempColumns = array(
	'tx_beuserprofilepic_profile_pic' => array(		
		'exclude' => 0,		
		'label' => 'LLL:EXT:be_user_profilepic/locallang_db.xml:be_users.tx_beuserprofilepic_profile_pic',		
		'config' => array(
			'type' => 'group',
			'internal_type' => 'file',
			'allowed' => $GLOBALS['TYPO3_CONF_VARS']['GFX']['imagefile_ext'],	
			'max_size' => $GLOBALS['TYPO3_CONF_VARS']['BE']['maxFileSize'],	
			'uploadfolder' => 'uploads/tx_beuserprofilepic',
			'show_thumbs' => 1,	
			'size' => 1,	
			'minitems' => 0,
			'maxitems' => 1,
		)
	),
);


t3lib_div::loadTCA('be_users');
t3lib_extMgm::addTCAcolumns('be_users',$tempColumns,1);
t3lib_extMgm::addToAllTCAtypes('be_users','tx_beuserprofilepic_profile_pic;;;;1-1-1');


if (TYPO3_MODE === 'BE')	{
	t3lib_extMgm::insertModuleFunction(
		'user_task',		
		'tx_beuserprofilepic_modfunc1',
		t3lib_extMgm::extPath($_EXTKEY).'modfunc1/class.tx_beuserprofilepic_modfunc1.php',
		'LLL:EXT:be_user_profilepic/locallang_db.xml:moduleFunction.tx_beuserprofilepic_modfunc1'
	);
}
?>

modfunc1/class.tx_beuserprofilepic_modfunc1.php
<?php
/***************************************************************
 *  Copyright notice
 *
 *  (c) 2012 Anchu <ccoollh@gmail.com>
 *  All rights reserved
 *
 *  This script is part of the TYPO3 project. The TYPO3 project is
 *  free software; you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation; either version 2 of the License, or
 *  (at your option) any later version.
 *
 *  The GNU General Public License can be found at
 *  http://www.gnu.org/copyleft/gpl.html.
 *
 *  This script is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  This copyright notice MUST APPEAR in all copies of the script!
 ***************************************************************/



/**
 * Module extension (addition to function menu) 'Add profile picture' for the 'be_user_profilepic' extension.
 *
 * @author	Anchu <ccoollh@gmail.com>
 * @package	TYPO3
 * @subpackage	tx_beuserprofilepic
 */
class tx_beuserprofilepic_modfunc1 extends mod_user_task {
					/**
					 * Makes the content for the overview frame...
					 *
					 * @return	HTML
					 */
					function overview_main()	{
						$icon = '<img src="'.$this->backPath.t3lib_extMgm::extRelPath("be_user_profilepic").'ext_icon.gif" width=18 height=16 class="absmiddle">';
						$content = $this->mkMenuConfig($icon.$this->headLink(tx_beuserprofilepic_modfunc1,1),'',$this->overviewContent());

						return $content;
					}

					/**
					 * Main method
					 *
					 * @return	HTML
					 */
					function main() {
						global $BE_USER,$LANG,$BACK_PATH,$TCA_DESCR,$TCA,$CLIENT,$TYPO3_CONF_VARS;

						return $this->mainContent();
					}

					/**
					 * Returns content in overview frame
					 *
					 * @return	Content for overview frame
					 */
					function overviewContent()	{
						$content = 'Content in overview frame...';
						return '<a href="index.php?SET[function]=tx_beuserprofilepic_modfunc1"  onClick="this.blur();"><img src="'.$this->backPath.'gfx/edit2.gif" style="float: left;"></a><div><a href="index.php?SET[function]=tx_beuserprofilepic_modfunc1"  onClick="this.blur();">'.$content.'</a></div>';
					}

					/**
					 * Main content method
					 *
					 * @return	Main content for the module
					 */
					function mainContent()	{
						return "Content in main frame...";
					}
				}



if (defined('TYPO3_MODE') && isset($GLOBALS['TYPO3_CONF_VARS'][TYPO3_MODE]['XCLASS']['ext/be_user_profilepic/modfunc1/class.tx_beuserprofilepic_modfunc1.php'])) {
	include_once($GLOBALS['TYPO3_CONF_VARS'][TYPO3_MODE]['XCLASS']['ext/be_user_profilepic/modfunc1/class.tx_beuserprofilepic_modfunc1.php']);
}

?>